src/compiler/zx_calculus.jl

```diff
@@ -1,14 +1,72 @@
-using ZXCalculus
+using ZXCalculus, LightGraphs
+using ZXCalculus: qubit_loc
 
 function optimize!(ir::YaoIR)
```

Comment:
`optimize!` should use a compiler context to dispatch optimize methods, since zx calculus won't be the only optimization strategy and can be disabled in a low optimization context. I'll rewrite this part later.